sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/samplers/ParentBasedSampler.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.trace.samplers;
+
+import io.opentelemetry.api.common.ReadableAttributes;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import java.util.List;
+import java.util.Objects;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * A Sampler that uses the sampled flag of the parent Span, if present. If the span has no parent,
+ * this Sampler will use the "root" sampler that it is built with. See documentation on the {@link
+ * Builder} methods for the details on the various configurable options.
+ */
+@Immutable
+public class ParentBasedSampler implements Sampler {
```

Comment:
This is top level now instead of `class Samplers.ParentBased`.

Actually if `ParentBasedSamplerBuilder` is made top-level, this class could be made private.